### PR TITLE
[16.0][MOV] sale_by_packaging becomes sell_only_by_packaging

### PR DESCRIPTION
**Pull request by QuocDuong1306**
This module feature was extracted from the original [sale-workflow/sale_by_packaging](https://github.com/oca/sale-workflow/tree/14.0/sale_by_packaging) module.

This module supports 2 features based on sale_by_packaging v14:

- Sell only by packaging: On product template model, this checkbox restricts
sales of these products if no packaging is selected on the sale order line.
If no packaging is selected, it will either be auto-assigned if the quantity
on the sale order line matches a packaging quantity or an error will be raised.

- Force sale quantity (on the packaging): force rounds up the quantity during
creation/modification of the sale order line with the factor set on the packaging.

Change:
- `sell_only_by_packaging` is now company dependent

Depend on:
- https://github.com/OCA/product-attribute/pull/1395

Base on:
- https://github.com/OCA/sale-workflow/pull/2570

### CODE REVIEW

**Review by jbaudoux — APPROVED** [review]
LG, not tested

**Review by marielejeune — APPROVED** [review]
LGTM (code review  + functional testing)

Just a comment: as this code comes from `sale_by_packaging`, I would keep copyright's year (see OCA recommandation here: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0#tasks-not-to-do-in-the-migration) and keep previous contributors in the contributors' list.

File: test-requirements.txt [hunk 1]
```diff
@@ -1 +1 @@
-odoo_test_helper
```
Comment (marielejeune) [inline test-requirements.txt]: Don't remove `odoo_test_helper` :)

Comment (QuocDuong1306) [reply test-requirements.txt]: Sorry, I have updated, could u take a look, thanks

Comment (marielejeune) [reply test-requirements.txt]: Perfect, thanks!

File: sell_only_by_packaging/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    "summary": "Manage sale of packaging",
+    "version": "16.0.1.0.0",
+    "development_status": "Alpha",
+    "category": "Warehouse Management",
+    "website": "https://github.com/OCA/sale-workflow",
+    "author": "Camptocamp, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "application": False,
+    "installable": True,
+    "depends": ["product_packaging_level_sellable", "sale_stock"],
```
Comment (marielejeune) [inline sell_only_by_packaging/__manifest__.py:14]: @QuocDuong1306 Can you update module name here please?

Comment (marielejeune) [inline sell_only_by_packaging/__manifest__.py]:
```suggestion
    "depends": ["product_packaging_level_salable", "sale_stock"],
```

File: sell_only_by_packaging/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+{
+    "name": "Sell Only By Packaging",
+    "summary": "Manage sale of packaging",
+    "version": "16.0.1.0.0",
+    "development_status": "Alpha",
+    "category": "Warehouse Management",
+    "website": "https://github.com/OCA/sale-workflow",
+    "author": "Camptocamp, Odoo Community Association (OCA)",
```
Comment (jbaudoux) [inline sell_only_by_packaging/__manifest__.py]:
```suggestion
    "author": "Camptocamp, BCIM, Odoo Community Association (OCA)",
```

File: sell_only_by_packaging/readme/CONTRIBUTORS.rst [hunk 4]
```diff
@@ -0,0 +1,6 @@
+* Akim Juillerat <akim.juillerat@camptocamp.com>
```
Comment (jbaudoux) [inline sell_only_by_packaging/readme/CONTRIBUTORS.rst:1]:
```suggestion
* Akim Juillerat <akim.juillerat@camptocamp.com>
* Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
```

**Comment by QuocDuong1306** [comment]
Hi @jbaudoux , I updated

**Comment by simahawk** [comment]
@QuocDuong1306 @jbaudoux don't we need a post install hook to migrate data from the old module?

File: sell_only_by_packaging/models/sale_order_line.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+class SaleOrderLine(models.Model):
+
+    _inherit = "sale.order.line"
+
+    @api.constrains(
+        "product_id", "product_packaging_id", "product_packaging_qty", "product_uom_qty"
+    )
+    def _check_product_packaging_sell_only_by_packaging(self):
+        for line in self:
+            if not line.product_id.sell_only_by_packaging:
```
Comment (santostelmo) [inline sell_only_by_packaging/models/sale_order_line.py]:
```suggestion
            if not line.product_id.sell_only_by_packaging or not line.product_uom_qty:
```
@QuocDuong1306 IMO there's no point to check lines without any quantity assigned. If we are not able to assign 0 to the `product_uom_qty` we get stuck in many circumstances (specially when the order is already confirmed).

**Review by santostelmo — CHANGES_REQUESTED** [review]

File: sell_only_by_packaging/models/sale_order_line.py (hunk 5 above)
Comment (santostelmo) [reply sell_only_by_packaging/models/sale_order_line.py]: @QuocDuong1306 Thank you for the change

**Review by santostelmo — APPROVED** [review]

**Review by cyrilmanuel — APPROVED** [review]
not an expert LGTM

**Pull request merged**